homeassistant/components/powerwall/sensor.py

```diff
@@ -281,3 +287,126 @@ def native_value(self) -> float | None:
+    ),
+    PowerwallSensorEntityDescription[BatteryResponse, float](
+        key="instant_current",
+        translation_key="instant_current",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.CURRENT,
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        entity_registry_enabled_default=False,
+        value_fn=lambda battery_data: round(battery_data.i_out, 1),
+    ),
+    PowerwallSensorEntityDescription[BatteryResponse, int](
+        key="instant_power",
+        translation_key="instant_power",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.POWER,
+        native_unit_of_measurement=UnitOfPower.WATT,
+        value_fn=lambda battery_data: battery_data.p_out,
+    ),
+    PowerwallSensorEntityDescription[BatteryResponse, float](
+        key="battery_export",
+        translation_key="battery_export",
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        suggested_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        suggested_display_precision=0,
+        value_fn=lambda battery_data: battery_data.energy_discharged,
+    ),
+    PowerwallSensorEntityDescription[BatteryResponse, float](
+        key="battery_import",
+        translation_key="battery_import",
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        suggested_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        suggested_display_precision=0,
+        value_fn=lambda battery_data: battery_data.energy_charged,
+    ),
+    PowerwallSensorEntityDescription[BatteryResponse, int](
+        key="battery_remaining",
+        translation_key="battery_remaining",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.ENERGY_STORAGE,
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        suggested_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        suggested_display_precision=1,
+        value_fn=lambda battery_data: battery_data.energy_remaining,
+    ),
+    PowerwallSensorEntityDescription[BatteryResponse, float](
+        key="charge",
+        translation_key="charge",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        suggested_display_precision=0,
+        value_fn=_get_battery_charge,
+    ),
+    PowerwallSensorEntityDescription[BatteryResponse, str](
+        key="grid_state",
+        translation_key="grid_state",
+        device_class=SensorDeviceClass.ENUM,
+        options=[state.value.lower() for state in GridState],
+        value_fn=lambda battery_data: battery_data.grid_state.value.lower(),
+    ),
+]
```

Comment:
It would be nicer to move this to the top of the file with the other constant and helper